### PR TITLE
Split API controller tests

### PR DESCRIPTION
**Pull request by sjanota**
**Description**

Changes proposed in this pull request:

- split api-controller tests into two parts: integration with istio and api-controller component tests
- integration tests with istio are more heavy-weight tests, thus there are less scenarios which checks if api-controller creates CRs understandable by istio
- component tests test api-controller in isolation, which means they verify only direct output of the api-controller (created CRs) and not the effect they have on istio. They are lighter and there is more of them. They are meant to check api-controller internals easier. 

What's the gain? Tests are more reliable as bulk of functionality is tested without interaction with istio and execute three times faster.

**Related issue(s)**

### CODE REVIEW

File: tests/api-controller-acceptance-tests/apicontroller/component_test.go [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+	return strings.Join(deep.Equal(actual, expected[0]), "\n")
+}
+
+func (componentTestContext) cleanUpApi(kymaClient *kyma.Clientset, api * kymaApi.Api, t *testing.T, allowMissing bool) {
+	if api == nil {
+		return
+	}
+	err := kymaClient.GatewayV1alpha2().Apis(namespace).Delete(api.Name, &metav1.DeleteOptions{})
+	if !allowMissing && err != nil {
+		t.Fatalf("Cannot clean up API: %s", err)
```
Comment (piotrmsc) [inline tests/api-controller-acceptance-tests/apicontroller/component_test.go]: Would be nice to have here the name or id of the API

File: tests/api-controller-acceptance-tests/apicontroller/integration_test.go [hunk 2]
```diff
@@ -83,170 +72,60 @@ func TestSpec(t *testing.T) {
...
 
 	return &kymaApi.Api{
 		ObjectMeta: metav1.ObjectMeta{
 			Namespace: namespace,
 			Name:      fmt.Sprintf("sample-app-api-%s", testId),
 		},
 		Spec: kymaApi.ApiSpec{
-			Hostname: hostnameFor(testId, domainName, hostWithDomain),
+			Hostname: ctx. hostnameFor(testId, domainName, hostWithDomain),
 			Service: kymaApi.Service{
```
Comment (piotrmsc) [inline tests/api-controller-acceptance-tests/apicontroller/integration_test.go]: `ctx.hostnameFor(..)`

**Comment by sjanota** [comment]
Awaits #1597 to be ready to merge.

**Comment by sjanota** [comment]
CLA is broken, will recreate PR

**Pull request closed without merging**